zebra-rpc/src/server/http_request_compatibility.rs

```diff
@@ -141,4 +161,30 @@ impl FixHttpRequestMiddleware {
             );
         }
     }
+
+    /// Check if the request is authenticated.
+    pub fn check_credentials(&self, headers: &header::HeaderMap) -> bool {
+        if !self.0.enable_cookie_auth {
+            return true;
+        }
+        headers
+            .get(header::AUTHORIZATION)
+            .and_then(|auth_header| auth_header.to_str().ok())
+            .and_then(|auth| auth.split_whitespace().nth(1))
+            .and_then(|token| URL_SAFE.decode(token).ok())
+            .and_then(|decoded| String::from_utf8(decoded).ok())
+            .and_then(|decoded_str| {
+                decoded_str
+                    .split(':')
+                    .nth(1)
+                    .map(|password| password.to_string())
+            })
+            .map_or(false, |password| {
+                if let Ok(cookie_password) = cookie::get(self.0.cookie_dir.clone()) {
```

Comment:
We're still reading the cookie from the disk on each RPC request.

Reply:
https://github.com/ZcashFoundation/zebra/pull/8900/commits/43131b8a5645f76c0d3b453a0099d9a7061b5f5a